### PR TITLE
Validate max length of company name

### PR DESCRIPTION
**Pull request by jie-d**
Changes:
- Added max length constraint to company name (defined [here](https://sourcegraph.prod.tableautools.com/teams/near/-/blob/modules/connectors/tabconnectorsbase/public/infrastructure/IConnectorPlugin.h#L7:22); already [enforced](https://sourcegraph.prod.tableautools.com/teams/near/-/blob/modules/connectors/tabmixins/main/infrastructure/ConnectorPluginValidator.cpp#L62) at load time so this change shouldn't break any existing gallery connectors)
- Renamed type `NonEmptyString-ST` to `NonEmptyCompanyName-ST`. It was provided a generic name, however it now has constraint specifically applies to company name so renamed it (currently this type is only used to validate company name).

Testing:
- Change the company name in \samples\plugins\postgres_jdbc\manifest.xml to an empty string and try packaging the connector
Found error in \connector-packager\packaging_logs.txt:
```
failed validating '' with XsdMinLengthFacet(value=1, fixed=False):
Reason: attribute name='': value length cannot be lesser than 1
```
- Change the company name in \samples\plugins\postgres_jdbc\manifest.xml to a string with length 28 and try packaging the connector
Found error in \connector-packager\packaging_logs.txt:
```
failed validating '1234567123456712345671234567' with XsdMaxLengthFacet(value=24, fixed=False):
Reason: attribute name='1234567123456712345671234567': value length cannot be greater than 24
```

To-Do: might need to add `NonEmptyCompanyName-ST` to the [root xsd](https://sourcegraph.prod.tableautools.com/teams/near/-/blob/modules/XSD/ConnectorPluginManifest-root.xsd) once I understand why `NonEmptyString-ST` is not there already.

### CODE REVIEW

**Review by rosswbrown — COMMENTED** [review]
Add a test case for empty string and >24 company name in [test_xsd_validator.py](https://github.com/tableau/connector-plugin-sdk/blob/master/connector-packager/tests/test_xsd_validator.py)

File: connector-packager/tests/test_xsd_validator.py [hunk 1]
```diff
@@ -254,3 +254,17 @@ def test_all_fields_in_normalizer(self):
...
+    def test_validate_company_name_length(self):
+        xml_violations_buffer = []
+        file_to_test = ConnectorFile("manifest.xml", "manifest")
+
+        print("Test that company name with length less than 1 is invalidated")
+        test_file = TEST_FOLDER / "company_name_length_validation/min/manifest.xml"
+        self.assertFalse(validate_single_file(file_to_test, test_file, xml_violations_buffer, dummy_properties),
+                         "Empty company name marked as valid")
+
+        print("Test that ompany name with length greater than 24 is invalidated")
```
Comment (pvanderknyff) [inline connector-packager/tests/test_xsd_validator.py:267]:
```suggestion
        print("Test that company name with length greater than 24 is invalidated")
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 # Tableau Connector SDK Changelog
+## 2021-09-02
+### Changed
+- Validate max length of `name` in Company-G in `connector_plugin_manifest_latest.xsd`
```
Comment (pvanderknyff) [inline CHANGELOG.md:4]: Thank you for being the only person to remember to do this.

**Review by pvanderknyff — APPROVED** [review]
LGTM, apart from one typo

File: connector-packager/tests/test_xsd_validator.py (hunk 1 above)
Comment (jie-d) [reply connector-packager/tests/test_xsd_validator.py:267]: Nice catch!

File: CHANGELOG.md (hunk 2 above)
Comment (jie-d) [reply CHANGELOG.md:4]: Lol

**Review by rosswbrown — APPROVED** [review]
